jax/_src/ops/special.py

```diff
@@ -95,3 +97,22 @@ def logsumexp(a: ArrayLike, axis: Axis = None, b: ArrayLike | None = None,
     with jax.debug_nans(False):
       out = jnp.where(sign < 0, jnp.array(np.nan, dtype=out.dtype), out)
   return out
+
+@logsumexp.defjvp
+def _logsumexp_jvp(axis, keepdims, return_sign, where, primals, tangents):
+  a, b = primals
+  a_dot, b_dot = tangents
+  out = logsumexp(a, axis, b, keepdims, return_sign, where)
+
+  if return_sign:
+    out, sign = out
+
+  if b is None:
+    b = jnp.ones_like(a)
+  out_dot = jnp.exp(
+      logsumexp(a, axis, b * a_dot + b_dot, keepdims, where) - out)
```

Comment:
This expression looks a bit strange to me – each element of `out_dot` should reflect the gradient of the corresponding elements of `a_dot` and `b_dot`, so summing over them doesn't look right to me.

Reply:
I think we do need to sum along `axis`, otherwise the shapes don't add up: `b * a_dot + b_dot` has the same shape as `a`, but the expected `out_dot` shape should be the same as `out`, i.e. lower rank than `a`.